services/apps/alcs/src/portal/application-submission/submission-status/application-submission-status.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { ServiceNotFoundException } from '../../../../../../libs/common/src/exceptions/base.exception';
+import { ApplicationSubmission } from '../application-submission.entity';
+import { SubmissionStatusType } from './submission-status-type.entity';
+import { SUBMISSION_STATUS } from './submission-status.dto';
+import { ApplicationSubmissionToSubmissionStatus } from './submission-status.entity';
+
+// FIXME instead of new Date it should be utc date. Also all comparisons between dates from UI should happen in utc
+@Injectable()
+export class ApplicationSubmissionStatusService {
+  constructor(
+    @InjectRepository(ApplicationSubmissionToSubmissionStatus)
+    private statusesRepository: Repository<ApplicationSubmissionToSubmissionStatus>,
+    @InjectRepository(SubmissionStatusType)
+    private submissionStatusTypeRepository: Repository<SubmissionStatusType>,
+    @InjectRepository(ApplicationSubmission)
+    private applicationSubmissionRepository: Repository<ApplicationSubmission>,
+  ) {}
+
+  async setInitialStatuses(submissionUuid: string) {
+    const statuses = await this.submissionStatusTypeRepository.find();
+    const newStatuses: ApplicationSubmissionToSubmissionStatus[] = [];
+
+    for (const status of statuses) {
+      const newStatus = new ApplicationSubmissionToSubmissionStatus({
+        submissionUuid: submissionUuid,
+        statusTypeCode: status.code,
+      });
+
+      if (newStatus.statusTypeCode === SUBMISSION_STATUS.IN_PROGRESS) {
+        const date = new Date();
+        date.setHours(0, 0, 0, 0);
+        newStatus.effectiveDate = date;
+      }
+
+      newStatuses.push(newStatus);
+    }
+
+    return await this.statusesRepository.save(newStatuses);
+  }
+
+  async setStatusDate(
+    submissionUuid: string,
+    statusTypeCode: string,
+    effectiveDate?: Date | null,
```

Comment:
Would it be cleaner if you defaulted this? effectiveDate: Date | null = new Date()?